src/Microsoft.DotNet.Tools.Compiler.Native/Program.cs

```diff
@@ -91,6 +82,24 @@ private static int RunMcg(NativeCompileSettings config)
             mcgArgs.Add("--outputpath");
             mcgArgs.Add(outPath);
 
+            var ilSdkPath = Path.Combine(config.IlcSdkPath, "sdk");
+            foreach (string refPath in Directory.EnumerateFiles(ilSdkPath, "*.dll"))
+            {
+
+                mcgArgs.Add("--r");
+                mcgArgs.Add(refPath);
+            }
+
+            foreach (string refPath in Directory.EnumerateFiles(config.AppDepSDKPath, "*.dll"))
+            {
+                // System.Runtime.Extensions define an internal type called System.Runtime.InteropServices.Marshal which 
+                // conflicts with Marshal from S.P.Interop , we don't need System.Runtime.Extensions anyways,skip it.
+                if (refPath.Contains("System.Runtime.Extensions.dll")) continue;
```

Comment:
Oh ,you are specifically asking about if ( contains check.)  ( i still haven't figured how git comments work) . yes , it's a point in time thing , we will rename the Marshal in  System.Runtime.Extensions.dll.